### PR TITLE
Add support for uv sets 3 to 6 in all materials

### PR DESCRIPTION
**Pull request by Popov72**
"all materials": standard, PBR and node materials.

### CODE REVIEW

File: src/Materials/Node/nodeMaterial.ts [hunk 1]
```diff
@@ -750,15 +755,19 @@ export class NodeMaterial extends PushMaterial {
...
         let oldTangent = defines["TANGENT"];
-        let oldUV1 = defines["UV1"];
 
         defines["NORMAL"] = mesh.isVerticesDataPresent(VertexBuffer.NormalKind);
 
         defines["TANGENT"] = mesh.isVerticesDataPresent(VertexBuffer.TangentKind);
 
-        defines["UV1"] = mesh.isVerticesDataPresent(VertexBuffer.UVKind);
+        let uvChanged = false;
+        for (let i = 1; i <= Constants.MAX_SUPPORTED_UV_SETS; ++i) {
```
Comment (deltakosh) [inline src/Materials/Node/nodeMaterial.ts:764]: love that constant!

File: src/Shaders/ShadersInclude/pbrFragmentSamplersDeclaration.fx [hunk 2]
```diff
@@ -1,194 +1,33 @@
...
-    #if REFLECTANCEDIRECTUV == 1
-        #define vReflectanceUV vMainUV1
-    #elif REFLECTANCEDIRECTUV == 2
-        #define vReflectanceUV vMainUV2
-    #else
-        varying vec2 vReflectanceUV;
-    #endif
-    uniform sampler2D reflectanceSampler;
-#endif
+#include<samplerFragmentDeclaration>(_DEFINENAME_,ALBEDO,_VARYINGNAME_,Albedo,_SAMPLERNAME_,albedo)
```
Comment (deltakosh) [inline src/Shaders/ShadersInclude/pbrFragmentSamplersDeclaration.fx:1]: This is smart!

**Review by deltakosh — APPROVED** [review]

**Review by deltakosh — APPROVED** [review]

File: src/Shaders/ShadersInclude/samplerVertexImplementation.fx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#if defined(_DEFINENAME_) && _DEFINENAME_DIRECTUV == 0
+	if (v_INFONAME_ == 0.)
+	{
+		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uvUpdated, 1.0, 0.0));
+	}
```
Comment (sebavan) [inline src/Shaders/ShadersInclude/samplerVertexImplementation.fx:5]: Could we add #ifdef around them as we know the UVx from the #define ?

**Comment by sebavan** [comment]
This looks so good to me :-) Thanks a ton

File: src/Shaders/ShadersInclude/samplerVertexImplementation.fx (hunk 3 above)
Comment (sebavan) [reply src/Shaders/ShadersInclude/samplerVertexImplementation.fx:5]: on all but the first one :-)

Comment (Popov72) [reply src/Shaders/ShadersInclude/samplerVertexImplementation.fx:5]: What do you mean? I don't understand, can you give an example?

File: src/Shaders/ShadersInclude/samplerVertexImplementation.fx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv3, 1.0, 0.0));
+	}
+	else if (v_INFONAME_ == 3.)
+	{
+		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv4, 1.0, 0.0));
+	}
+	else if (v_INFONAME_ == 4.)
+	{
+		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv5, 1.0, 0.0));
```
Comment (sebavan) [inline src/Shaders/ShadersInclude/samplerVertexImplementation.fx:20]:
```suggestion
#ifdef UV1
	else if (v_INFONAME_ == 1.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv2, 1.0, 0.0));
	}
#endif
#ifdef UV2
	else if (v_INFONAME_ == 2.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv3, 1.0, 0.0));
	}
#endif
#ifdef UV3
	else if (v_INFONAME_ == 3.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv4, 1.0, 0.0));
	}
#endif
```

File: src/Shaders/ShadersInclude/samplerVertexImplementation.fx (hunk 3 above)
Comment (sebavan) [reply src/Shaders/ShadersInclude/samplerVertexImplementation.fx:5]:
```suggestion
#ifdef UV1
	else if (v_INFONAME_ == 1.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv2, 1.0, 0.0));
	}
#endif
#ifdef UV2
	else if (v_INFONAME_ == 2.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv3, 1.0, 0.0));
	}
#endif
#ifdef UV3
	else if (v_INFONAME_ == 3.)
	{
		v_VARYINGNAME_UV = vec2(_MATRIXNAME_Matrix * vec4(uv4, 1.0, 0.0));
	}
#endif
```

Comment (sebavan) [reply src/Shaders/ShadersInclude/samplerVertexImplementation.fx:5]: wondering if that makes sense ?

**Review by sebavan — APPROVED** [review]

**Comment by sebavan** [comment]
@Popov72 I guess we should mention this one in the what s new section :-)

**Comment by deltakosh** [comment]
It is a major update

**Pull request merged**